File: src/main/java/july_2025/ExcelSheet.java
package july_2025;

public class ExcelSheet {

    public static void main(String[] args) {
        String columnTitle = "ZY";

        ExcelSheet excelSheet = new ExcelSheet();
        int result = excelSheet.titleToNumber(columnTitle);
        System.out.println(result);
    }

    // O(n) time | O(1) space
    public int titleToNumber(String columnTitle) {
        int sum = 0;
        int pow = 1;
        for (int i = columnTitle.length() - 1; i >= 0; i--) {
            char c = columnTitle.charAt(i);

            int val = c - 'A' + 1;
            sum = sum + val * pow;
            pow *= 26;
        }
        return sum;
    }

}

File: python/second/excel_sheet.py
class Solution(object):
    # O(n) time | O(1) space
    def titleToNumber(self, columnTitle):
        """
        :type columnTitle: str
        :rtype: int
        """
        sum = 0
        pow = 1

        for i in range(len(columnTitle) - 1, -1, -1):
            c = ord(columnTitle[i]) - ord("A") + 1

            sum += pow * c
            pow *= 26
        
        return sum

solution = Solution()
print(solution.titleToNumber("ZY"))